src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -238,9 +238,10 @@ class ObjectMonitor : public CHeapObj<mtInternal> {
 
   bool is_busy() const {
     // TODO-FIXME: assert _owner == null implies _recursions = 0
-    intptr_t ret_code = _waiters | intptr_t(_cxq) | intptr_t(_EntryList);
-    if (contentions() > 0) {
-      ret_code |= contentions();
+    intptr_t ret_code = intptr_t(_waiters) | intptr_t(_cxq) | intptr_t(_EntryList);
+    int cnts = contentions(); // read once
+    if (cnts > 0) {
+      ret_code |= intptr_t(cnts);
```

Comment:
I know it has nothing to do with your patch, but I am just curious. IIUC this is a strange way of saying "if A!=0 || B!=0 || C!=0..." . Is this for performance reasons? Also, why the "if (cnts > 0)", would that not be superfluous?

Reply:
I think this code exists like this from when is_busy returned an uintptr_t.  It should be nicely rewritten with if != 0s now and likely would have better performance.  I won't do it with this change because these fields are volatile and I think should have Atomic::loads, which would lead to a lot of discussion that should happen independently.  I'll file an RFE.